core/storage/user/gae_models_test.py

```diff
@@ -242,3 +242,51 @@ def test_get_multi_failure(self):
             SKILL_IDS)
 
         self.assertEqual(retrieved_object, [None, None])
+
+
+class UserContributionsScoringModelTests(test_utils.GenericTestBase):
+    """Tests for UserContributionScoringModel."""
+
+    def test_create_model(self):
+        user_models.UserContributionScoringModel.create('user1', 'category1', 1)
+        scoreModels = (user_models.UserContributionScoringModel
+                       .get_all_scores_of_user('user1'))
+        self.assertEqual(len(scoreModels), 1)
+        self.assertEqual(scoreModels[0].id, 'category1.user1')
+        self.assertEqual(scoreModels[0].user_id, 'user1')
+        self.assertEqual(scoreModels[0].score_category, 'category1')
+        self.assertEqual(scoreModels[0].score, 1)
+
+    def test_create_failure(self):
+        user_models.UserContributionScoringModel.create('user1', 'category1', 1)
+        with self.assertRaisesRegexp(
+            Exception, 'There is already an entry with the given id:'
+                       ' category1.user1'):
+            user_models.UserContributionScoringModel.create(
+                'user1', 'category1', 2)
+
+    def test_get_all_users_with_score_above_minimum_for_category(self):
+        user_models.UserContributionScoringModel.create('user1', 'category1', 1)
+        user_models.UserContributionScoringModel.create(
+            'user2', 'category1', 21)
+        user_models.UserContributionScoringModel.create(
+            'user3', 'category1', 11)
+        user_models.UserContributionScoringModel.create(
+            'user4', 'category1', 11)
+        user_models.UserContributionScoringModel.create(
+            'user1', 'category2', 11)
+        user_models.UserContributionScoringModel.create('user2', 'category2', 1)
+        user_models.UserContributionScoringModel.create('user3', 'category2', 1)
+        user_models.UserContributionScoringModel.create('user4', 'category2', 1)
+
+        score_models = (user_models.UserContributionScoringModel
+                        .get_all_users_with_score_above_minimum_for_category(
+                            'category1'))
+
+        self.assertEqual(len(score_models), 3)
```

Comment:
Consider verifying that these are also the correct models.

Reply:
Done